### PR TITLE
Fix 10519 UtilityCost:Tariff with Electric:Facility showing zero

### LINKED ISSUE #10519: Tariffs using Electricity:Facility have zero annual energy costs
Issue overview
--------------
When running the example file StripMallZoneEvapCooler.idf but, changing it to an annual simulation, the Tariff reports in the tabular output file show zero costs for energy. This is due to the use of Electricity:Facility meter on those UtilityCost:Tariff being the first meter defined. 

The fix is to change the following line to test for -1 instead of 0:

https://github.com/NREL/EnergyPlus/blob/4c9271b7d07e4383284eeb30ea06d09aed329cbb/src/EnergyPlus/EconomicTariff.cc#L2579

Also need to add a CI test that includes an annual simulation for a file that contains a UtilityCost:Tariff so this type of bug is not missed in the future.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus 24.1
 - Helpdesk ticket number: 16659 (use the meter ElectricityPurchased:Facility as a workaround)

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by JasonGlazer**
Pull request overview
---------------------
 - Fixes #10519 
 - 

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by JasonGlazer** [comment]
This is ready for review. The fix itself was trivial but important since it would impact any file using tariffs with Electric:Facility meter. So I added some extra unit tests to increase coverage for the EconomicTariff.cc file.

File: src/EnergyPlus/EconomicTariff.cc [hunk 1]
```diff
@@ -2576,7 +2576,7 @@ void GatherForEconomics(EnergyPlusData &state)
         for (int iTariff = 1; iTariff <= state.dataEconTariff->numTariff; ++iTariff) {
             auto &tariff = state.dataEconTariff->tariff(iTariff);
             // if the meter is defined get the value
-            if (tariff.reportMeterIndx != 0) {
+            if (tariff.reportMeterIndx != -1) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/EconomicTariff.cc:2579]: Off by one.  So cliché.   :D :D :D

File: testfiles/_5ZoneAirCooled_annual.idf [hunk 2]
```diff
@@ -3561,3 +3561,39 @@
...
+    Annual,                  !- Season
+    DemandCharges,           !- Category Variable Name
+    ,                        !- Remaining Into Variable
+    ,                        !- Block Size Multiplier Value or Variable Name
+    20,                      !- Block Size 1 Value or Variable Name
+    5.38,                    !- Block 1 Cost per Unit Value or Variable Name
+    80,                      !- Block Size 2 Value or Variable Name
+    4.23,                    !- Block 2 Cost per Unit Value or Variable Name
+    remaining,               !- Block Size 3 Value or Variable Name
+    3.60;                    !- Block 3 Cost per Unit Value or Variable Name
```
Comment (Myoldmopar) [inline testfiles/_5ZoneAirCooled_annual.idf:3597]: Great to cover this condition in the test file.  :+1:

**Review by Myoldmopar — APPROVED** [review]
All good here!

**Comment by Myoldmopar** [comment]
There's no need to wait on anything else here, this is a great minimal fix and nice test coverage.  Thanks @JasonGlazer

**Pull request merged**